### ISSUE TITLE
Top level pyro object should sublcass nn.Module and not PyroModule

### ISSUE THREAD

**Issue by adamgayoso** [cellarium-ai/cellarium-ml#39]
Just wanted to flag this. I can't recall the reason at the moment (could be namespace collision), but it was advised to us from the Pyro team to use a torch `nn.Module` to wrap a pyro model and pyro guide. 

https://github.com/cellarium-ai/scvi-distributed/blob/d8d3280845192069d7792417ee612e8eb2be3b6e/scvid/module/_probabilisticpca.py#L15

In scvi-tools this is exactly what we do 

https://github.com/scverse/scvi-tools/blob/257a9f706f5c961fdff460d7a9836a9cab8bca15/scvi/module/base/_base_module.py#L295-L312

And it seems like best practice is to use PyroModules for each of the model and guide separately

**Comment by ordabayevy** [comment]
Thanks for the tips @adamgayoso! I agree that having the model and guide as separate `PyroModule`s is better approach. As to the `ProbabilisticPCAPyroModule` I think the neat solution would be to make it an [`ELBOModule`](https://docs.pyro.ai/en/stable/inference_algos.html#pyro.infer.elbo.ELBOModule) subclass (which is in turn a `nn.Module` subclass). Something like:

```py
class ProbabilisticPCA(ELBOModule):
    def __init__(self, ...):
        model = ProbabilisticPCAPyroModel(...)
        guide = ProbabilisticPCAPyroGuide(...)
        elbo = Trace_ELBO()
        super().__init__(model, guide, elbo)
```

Then `ProbabilisticPCA` can be passed in to the `PyroTrainingPlan` like in https://github.com/pyro-ppl/pyro/pull/3189:

```py
class PyroTrainingPlan(pl.LightningModule):
    def __init__(self, loss_fn: pyro.infer.elbo.ELBOModule, lr: float):
        super().__init__()
        self.loss_fn = loss_fn
        self.lr = lr

    def training_step(self, batch, batch_idx):
        args, kwargs = self.loss_fn._get_args_from_batch(batch)
        loss = self.loss_fn(*args, **kwargs)
        return loss

    def configure_optimizers(self):
        return torch.optim.Adam(self.loss_fn.parameters(), lr=self.lr)
```

One part that I'm stuck on is the learnable `self.mean_g` parameter that is shared between the model and guide. I'm not sure where is the best place to define it and how to make it shareable between separated model/guide pair. In Pyro parameters can be shared through the common `ParamStore` but here parameters are trained using `torch.optimizer` so not sure how that would work out.

**Comment by adamgayoso** [comment]
I see, we looked into `ElboModule` but it seemed like it was more for internal use. 

As for shared parameters, you can follow the paradigm used in the [scanvi pyro example](https://pyro.ai/examples/scanvi.html)

In particular they use a `nn.Module` and `pyro.module("scanvi", self)`

**Comment by ordabayevy** [comment]
Closed via #100
